docs/reference/module-api/module-overview.md

```diff
@@ -483,18 +483,354 @@ module.exports = {
+        const response = await _super(req);
+
+        if (Array.isArray(response.results)) {
+          // Adds a `resultLength` property on the response object.
+          response.resultLength = response.results.length;
+        }
+
+        return response;
+      }
+    };
+  }
+};
+```
+
+### `apiRoutes(self)`
+
+Add custom API routes. The `apiRoutes` function takes takes the module as an argument and must return an object with properties for the relevant [HTTP request method](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods), including `get`, `post`, `patch`, and `delete`. Each of those properties should be set to an object of functions.
+
+```javascript
+// modules/product/index.js
+module.exports = {
+  // ...
+  apiRoutes(self) {
+    return {
+      get: {
+        // GET /api/v1/product/cheapest
+        async cheapest(req) {
+          const product = await self.find(req).sort({
+            price: 1
+          }).toObject();
+          if (!product) {
+            // Browser receives a 404 error
+            throw self.apos.error('notfound', 'No products were found.');
+          }
+          // Response is a JSON object with a `product` property
+          return {
+            product
+          };
+        }
+      }
+    };
+  }
+};
+```
+
+#### Naming routes
+
+If route properties do *not* begin with a forward slash, they can be reached via the pattern: `/api/v1/` followed by the module name, a forward slash, and the API route handler's property name, e.g., `/api/v1/product/cheapest` for the `cheapest` route on the `product` module.
+
+Camel-case names will be converted to kebab case names for the URL: `cheapestOne` becomes `cheapest-one` in the route URL.
+
+You can also use a completely custom URL path by starting the name of the route with a forward slash (`/`). If you do so, nothing will be prefixed to it and it will not be converted to kebab case.
+
+```javascript
+// modules/product/index.js
+module.exports = {
+  // ...
+  apiRoutes(self) {
+    return {
+      get: {
+        // GET /api/v1/product/cheapest-one
+        async cheapestOne(req) {
+          // ...
+        },
+        // GET /my-api/cheapest
+        '/my-api/cheapest': async function (req) {
+          // ...
+        }
+      }
+    };
+  }
+};
+```
+
+#### Returning error codes
+
+You can return the `self.apos.error()` method in a route function to return specific error codes. Pass in one of several strings to set a specific error response code:
+
+| Error name | HTTP response code |
+| ---- | ---- |
+| `'invalid'` | 400 |
+| `'forbidden'` | 403 |
+| `'notfound'` | 404 |
+| `'required'` | 422 |
+| `'conflict'` | 409 |
+| `'locked'` | 409 |
+| `'unprocessable'` | 422 |
+| `'unimplemented'` | 501 |
+
+Passing a different value as the first argument to `self.apos.error()` will set the response code to 500.
+<!-- TODO: Link to the method's own documentation page when available for more. -->
```

Comment:
I'm holding off on full documentation of that here since it could be a big chunk. I'm going to link over to it once available instead. There'll be a sentence like "See more information on using the `error()` method."